xtask/src/release.rs

```diff
@@ -0,0 +1,154 @@
+use serde_json::json;
+use xshell::{cmd, read_file};
+
+use crate::{config, flags, project_root, Result};
+
+impl flags::Release {
+    /// Run the release command to effectively create a release.
+    pub(crate) fn run(self) -> Result<()> {
+        let config = self.get_config()?;
+
+        let version = self.get_version()?;
+        println!("Making GitHub release for {} {}…", self.name, version);
+
+        let changes = self.get_changes(&version)?;
+
+        let tag = format!("{}-{}", self.name, version);
+        let name = format!("{} {}", self.name, version);
+
+        cmd!("git tag").args(&["-a", &tag, "-m", &name, "-m", &changes]).secret(true).run()?;
```

Comment:
I mainly wanted to have the output in case there's an error, but I guess we can just forward the error. It's also for the user to see that something is happening.

Reply:
Okay, we can keep it like thta for this PR, it's easy to change either way.

Reply:
I have to admit though that showing `secret` is pretty useless. We don't even know what's happening…

> Why is tag part of the original cmd! instead of the item in the args list?

It's just that I started by putting everything in `cmd!` but it turns out interpolation doesn't work inside single quotes in the string so I had to use `args` instead. Anyway, I'll put it in args, it's more logical that way.

Reply:
Oh, of course, interpolation. I think you can actually use that here and leave out the quotes, IIRC `cmd!` makes sure that interpolated variables don't get expanded.

Reply:
Ah right, we can now. It's because when I started there wasn't the `name` variable so it couldn't handle the space between the variables.